demos/echo.rst

```diff
@@ -0,0 +1,93 @@
+========================
+OpenAMP Echo Test Sample
+========================
+
+.. _echo-test-intro:
+
+***************
+Echo Test Intro
+***************
+
+The echo test reference sample, as the name suggests, demonstrates OpenAMP :ref:`Interprocessor Communications (IPC)<ipc-work-label>` components by providing an echo application on a remote which simply returns (echoes) packets as they are received at an :ref:`RPmsg endpoint <rpmsg-endpoint>` from the host controller. The host controller then verifies the returned packet for integrity.
+
+..  image::  ../images/demos/echo-test-intro.svg
+
+.. _echo-test-components:
+
+********************
+Echo Test Components
+********************
+
+There are two applications involved in this demonstration.
+The :ref:`remote application<echo-test-remote-app>` runs as an echo service, which returns packets it receives on an :ref:`RPmsg endpoint <rpmsg-endpoint>`.
+The :ref:`host application<echo-test-host-app>` is the test application sending packets to the echo service and monitoring for their return.
+
+The underlying OpenAMP architectural components used by these applications are
+
+* :ref:`Remoteproc<overview-remoteproc-work-label>`
+* :ref:`Resource Table<overview-remoteproc-work-label>`
+* :ref:`RPMsg<overview-rpmsg-work-label>`
+* :ref:`Virtio<overview-rpmsg-work-label>`
+* :ref:`Libmetal<overview-proxy-libmetal-label>`
+
+The following architecture diagram shows the components involved in the demonstration.
+
+..  image::  ../images/demos/echo-test-components.svg
+
+The top-level control flow is shown in the following message diagram.
+
+..  image::  ../images/demos/echo-test-control-flow.svg
+
+.. _echo-test-remote-app:
+
+RPmsg Echo Remote Application
+=============================
+
+The remote application, rpmsg-echo, is the core of the demonstration. It is a simple application serving a :ref:`RPmsg endpoint <rpmsg-endpoint>` running as the main task on the remote processor, once loaded and started using :ref:`Remoteproc<overview-remoteproc-work-label>`.
+
+
+.. _echo-test-host-app:
+
+Echo Test Host Application
+==========================
+
+The echo_test application forms the host controller side of the demonstration. It repeatedly writes an increasing length payload of 0xA5's up to the maximum data size (packet size minus header) to the RPmsg endpoint. Following each packet send, it reads from the same endpoint and verifies the returned packet for correctness. The application will stop and report on the first corruption found.
+
+Echo Test Host Script
+=====================
+
+The host is also responsible for loading the firmware containing the :ref:`RPmsg Echo Remote Application<echo-test-remote-app>` and starting the remote processor using :ref:`Remoteproc<overview-remoteproc-work-label>`.
```

Comment:
In the echo test itself,  the host does not load the remote firmware. it tests only rpmsg protocol with the remoteproc virtio transport. here is more a prerequiste

Reply:
The docker demo has demo1A script and which has echo image > /sys/class/remoteproc/remoteproc0/firmware, and then sends start to .../state. That is what this "Host Script" section is referring to, rather then the echo_test application itself.
Clarified by adding reference to demo1A of Demo Docker Images